### PR TITLE
Fix magic path

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Would close https://github.com/iterative/setup-cml/issues/56; still doesn't work because of an esoteric path issue involving Windows, Emscripten, `libmagic` and a pinch of salt.

* Upstream fix: [1.3.12](https://github.com/npcz/magic/releases/tag/v1.3.12) https://github.com/npcz/magic/issues/10

### CODE REVIEW

**Comment by 0x2b3bfa0** [comment]
Currently works everywhere, except when `magic.mgc` is under the drive [`C:`](https://en.wikipedia.org/wiki/Drive_letter_assignment#Common_assignments) 🙃

File: src/utils.js [hunk 1]
```diff
@@ -28,11 +28,7 @@ const exec = async (command) => {
 
 const mimeType = async (opts) => {
   const { path, buffer } = opts;
-  FileMagic.magicFile = PATH.join(
-    __dirname,
-    '..',
-    'node_modules/@npcz/magic/dist/magic.mgc'
-  );
+  FileMagic.magicFile = require.resolve("@npcz/magic/dist/magic.mgc");
```
Comment (0x2b3bfa0) [inline src/utils.js:31]:
```suggestion
  FileMagic.magicFile = require.resolve('@npcz/magic/dist/magic.mgc');
```

**Comment by DavidGOrtega** [comment]
> Currently works everywhere, except when `magic.mgc` is under the drive [`C:`](https://en.wikipedia.org/wiki/Drive_letter_assignment#Common_assignments) 🙃

Why? The code is currently OS agnostic right? I mean 

```js
FileMagic.magicFile = PATH.join(
    __dirname,
    '..',
    'node_modules/@npcz/magic/dist/magic.mgc'
  );
```

**Comment by 0x2b3bfa0** [comment]
> _The code is currently OS agnostic right? I mean_
> ```javascript
> FileMagic.magicFile = PATH.join(
>     __dirname,
>     '..',
>     'node_modules/@npcz/magic/dist/magic.mgc'
>   );
> ```

If you mean using [`path.join`](https://nodejs.org/api/path.html#pathjoinpaths) just to pass a bunch of forward slashes as one of the arguments... 😈 Not quite agnostic.

**Comment by 0x2b3bfa0** [comment]
If any, the agnostic alternative would be this:

```javascript
FileMagic.magicFile = path.join(
  __dirname,
  '..',
  'node_modules',
  '@npcz',
  'magic',
  'dist',
  'magic.mgc'
);
```

I'm testing it also, just in case.

**Comment by DavidGOrtega** [comment]
i think in windows that does not matter. Like python

**Comment by 0x2b3bfa0** [comment]
😅 Today I learned that theay are fully equivalent, and you can even mix forward slashes with backslashes in the same path. 🤦🏼‍♂️

**Comment by 0x2b3bfa0** [comment]
This works as expected. Somebody is stripping the drive letter from paths.

```batch
cd C:\
vl2png /d/a/setup-cml/setup-cml/assets/vega-lite.json | cml publish
```

**Comment by DavidGOrtega** [comment]
I think that we should close this. In any case a similar code to this one is already merged in prod
@0x2b3bfa0

**Comment by 0x2b3bfa0** [comment]
Yes, the problem is still not fixed, but changes on this pull request are no longer relevant

**Pull request closed without merging**